<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Design and evaluate panels of genes or genomic regions • paneldesign</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Design and evaluate panels of genes or genomic regions">
<meta property="og:description" content="This R package provides functions for designing sequencing panels. A sequencing panel is a collection of genomic regions selected for targeted sequencing, such as cancer hotspot mutations. Given a cohort of individuals with associated genomic features (e.g., patients with somatic mutations), the typical objective is to design a panel that captures the largest number of individuals while using the smallest possible subset of features.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">paneldesign</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="paneldesign">paneldesign<a class="anchor" aria-label="anchor" href="#paneldesign"></a>
</h1></div>
<!-- badges: start -->

<p>This R package provides functions to design and evaluate seqsuencing panels. A sequencing panel is a collection of genomic regions selected for targeted sequencing, such as cancer hotspot mutations. Given a cohort of individuals with associated genomic features (e.g., patients with somatic mutations), the typical objective is to design a panel that captures the largest number of individuals while using the smallest possible subset of features.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="section level3">
<h3 id="install-bioconductor-dependencies">Install Bioconductor dependencies<a class="anchor" aria-label="anchor" href="#install-bioconductor-dependencies"></a>
</h3>
<p>This package depended on the Biocondoctor packages <a href="https://bioconductor.org/packages/release/bioc/html/GenomicRanges.html" class="external-link">GenomicRanges</a> and <a href="https://bioconductor.org/packages/release/bioc/html/GenomeInfoDb.html" class="external-link">GenomeInfoDb</a>. Install it by</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span> <span class="co"># if needed</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu">install</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"GenomicRanges"</span>, <span class="st">"GenomeInfoDb"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="install-this-package-from-github">Install this package from GitHub<a class="anchor" aria-label="anchor" href="#install-this-package-from-github"></a>
</h3>
<p>This R package is not yet on <a href="https://CRAN.R-project.org" class="external-link">CRAN</a> or <a href="https://www.bioconductor.org/" class="external-link">Biodonducotr</a>. Therefore, you have to install it form this GitHub repository.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("remotes")</span></span>
<span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"tron-bioinformatics/paneldesign"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="usage-example">Usage example<a class="anchor" aria-label="anchor" href="#usage-example"></a>
</h2>
<p>This is a basic example to show you how to use the package:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tron-bioinformatics.github.io/paneldesign">paneldesign</a></span><span class="op">)</span></span></code></pre></div>
<p>Given a data set of mutations in patients, the goal is to select mutations that are present in a maximum number of patients. We use a toy data set consisting of patients <code>p1</code>, <code>p2</code>, and <code>p3</code> of which each has 3 mutations.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mut_toy</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 6</span></span>
<span><span class="co">#&gt;   patient_id mut_id chr   start   end gene </span></span>
<span><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;</span></span>
<span><span class="co">#&gt; 1 p1         m01    1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 2 p1         m02    1      2000  2000 g1   </span></span>
<span><span class="co">#&gt; 3 p1         m03    2      3000  3000 g2   </span></span>
<span><span class="co">#&gt; 4 p2         m01    1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 5 p2         m04    2      4000  4000 g2   </span></span>
<span><span class="co">#&gt; 6 p2         m05    3      5000  5000 g3   </span></span>
<span><span class="co">#&gt; 7 p3         m02    1      2000  2000 g1   </span></span>
<span><span class="co">#&gt; 8 p3         m04    2      4000  4000 g2   </span></span>
<span><span class="co">#&gt; 9 p3         m06    2      6000  6000 g2</span></span></code></pre></div>
<div class="section level3">
<h3 id="select-mutations">Select mutations<a class="anchor" aria-label="anchor" href="#select-mutations"></a>
</h3>
<p>Here, we use a greedy algorithm to select a minimal set of mutations that covers all patients.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/select_greedy.html">select_greedy</a></span><span class="op">(</span><span class="va">mut_toy</span><span class="op">)</span></span>
<span><span class="co">#&gt; 100% covered by 2 sets.</span></span>
<span><span class="co">#&gt; # A tibble: 2 × 8</span></span>
<span><span class="co">#&gt;   mut_id     n n_samples  rank order n_cum coverage coverage_cum</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;  &lt;int&gt;     &lt;int&gt; &lt;dbl&gt; &lt;int&gt; &lt;int&gt;    &lt;dbl&gt;        &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 m01        2         3     1     1     2     66.7         66.7</span></span>
<span><span class="co">#&gt; 2 m02        1         3     2     2     3     33.3        100</span></span></code></pre></div>
<p>This results in two mutations <code>m01</code> and <code>m02</code>.</p>
</div>
<div class="section level3">
<h3 id="evaluate-given-panel-of-regions">Evaluate given panel of regions<a class="anchor" aria-label="anchor" href="#evaluate-given-panel-of-regions"></a>
</h3>
<p>Here we use the following toy genomic region set.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gr_toy</span></span>
<span><span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span></span>
<span><span class="co">#&gt;      seqnames    ranges strand</span></span>
<span><span class="co">#&gt;         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">#&gt;   r1        1      1000      *</span></span>
<span><span class="co">#&gt;   r3        2      3000      *</span></span>
<span><span class="co">#&gt;   -------</span></span>
<span><span class="co">#&gt;   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>First, we associate each region to patients by overlap with the mutation data</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reg_to_patient</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/panel_to_patient.html">panel_to_patient</a></span><span class="op">(</span><span class="va">gr_toy</span>, <span class="va">mut_toy</span><span class="op">)</span></span>
<span></span>
<span><span class="va">reg_to_patient</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 10</span></span>
<span><span class="co">#&gt;   reg_id reg_chr reg_start reg_end mut_id patient_id chr   start   end gene </span></span>
<span><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;       &lt;int&gt;   &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;</span></span>
<span><span class="co">#&gt; 1 r1     1            1000    1000 m01    p1         1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 2 r1     1            1000    1000 m01    p2         1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 3 r1     1            1000    1000 m01    p1         1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 4 r1     1            1000    1000 m01    p2         1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 5 r3     2            3000    3000 m03    p1         2      3000  3000 g2   </span></span>
<span><span class="co">#&gt; 6 &lt;NA&gt;   &lt;NA&gt;           NA      NA &lt;NA&gt;   p3         &lt;NA&gt;     NA    NA &lt;NA&gt;</span></span></code></pre></div>
<p>This results in the above-shown data set. The patient <code>p3</code> is not <em>covered</em> because there are no mutations in <code>p3</code> that overlap with the input regions. This results in ‘NA’ values for some region or mutation-specific columns.</p>
<p>Next, we evaluate each region from the panel for the patient coverage.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/eval_regions.html">eval_regions</a></span><span class="op">(</span><span class="va">reg_to_patient</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 2 × 15</span></span>
<span><span class="co">#&gt;   reg_id reg_chr reg_start reg_end n_region reg_size reg_size_cum n_patient</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;       &lt;int&gt;   &lt;int&gt;    &lt;int&gt;    &lt;dbl&gt;        &lt;dbl&gt;     &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 r1     1            1000    1000        1        1            1         2</span></span>
<span><span class="co">#&gt; 2 r3     2            3000    3000        2        1            2         1</span></span>
<span><span class="co">#&gt; # ℹ 7 more variables: n_patients_cum &lt;dbl&gt;, total_patients &lt;int&gt;,</span></span>
<span><span class="co">#&gt; #   percent_patients &lt;dbl&gt;, percent_patients_cum &lt;dbl&gt;, n_patient_gain &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   n_mut &lt;int&gt;, n_mut_cum &lt;dbl&gt;</span></span></code></pre></div>
<p>This assumes some order of the input regions by decreasing priority and in this way outputs additionally cumulative values such as <code>percent_patients_cum</code> which the cumulative percent of input patients that are covered by the current or any previous region. This is useful to decide where to <em>cut</em> the panel.</p>
<p>Lastly, we can get some overall performance metrics of the entire panel:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eval</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/eval_panel.html">eval_panel</a></span><span class="op">(</span><span class="va">reg_to_patient</span><span class="op">)</span></span>
<span><span class="va">eval</span></span>
<span><span class="co">#&gt; # A tibble: 1 × 9</span></span>
<span><span class="co">#&gt;   n_reg size_total total_patients n_patient percent_patient n_mut</span></span>
<span><span class="co">#&gt;   &lt;int&gt;      &lt;dbl&gt;          &lt;int&gt;     &lt;int&gt;           &lt;dbl&gt; &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1     2          2              3         2            66.7     2</span></span>
<span><span class="co">#&gt; # ℹ 3 more variables: mut_per_patient_median &lt;int&gt;, mut_per_patient_mean &lt;dbl&gt;,</span></span>
<span><span class="co">#&gt; #   mut_per_patient_df &lt;list&gt;</span></span></code></pre></div>
<p>This data set contains a nested data set with information on how many mutations each patient has represented on the panel.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">eval</span><span class="op">$</span><span class="va">mut_per_patient_df</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; # A tibble: 3 × 2</span></span>
<span><span class="co">#&gt;   patient_id mut_per_patient_n</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;                  &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 p1                         2</span></span>
<span><span class="co">#&gt; 2 p2                         1</span></span>
<span><span class="co">#&gt; 3 p3                         0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="structural-variants">Structural variants<a class="anchor" aria-label="anchor" href="#structural-variants"></a>
</h3>
<p>Structural variants (SVs) are represented as pairs of breakpoint coordinates. And to be covered by a panel region either a single (<code>sv_mode = "single"</code>) or both (<code>sv_mode = "both"</code>) breakpoints must be included in a region of the panel.</p>
<p>An toy example SV dataset might look like this</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sv_toy</span></span>
<span><span class="co">#&gt; # A tibble: 2 × 6</span></span>
<span><span class="co">#&gt;   patient_id mut_id bp1_chr bp1_pos bp2_chr bp2_pos</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt;     &lt;dbl&gt; &lt;chr&gt;     &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 p4         sv01   1          1000 1          1500</span></span>
<span><span class="co">#&gt; 2 p5         sv02   1          2000 2          2000</span></span></code></pre></div>
<p>We can evaluate a dataset with mutations and SVs by passing it as additional input to the <code><a href="reference/panel_to_patient.html">panel_to_patient()</a></code> function</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reg_to_patient</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/panel_to_patient.html">panel_to_patient</a></span><span class="op">(</span><span class="va">gr_toy</span>, <span class="va">mut_toy</span>, <span class="va">sv_toy</span>, sv_mode <span class="op">=</span> <span class="st">"single"</span><span class="op">)</span></span></code></pre></div>
<p>Now the <code>reg_to_patient</code> table includes additonal colums for the SV breakpoint coordinates and includes here an association of SV <code>sv01</code> from patient <code>p4</code> with region <code>r1</code>. In contrast patient <code>p5</code>is not covered by the panel.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">reg_to_patient</span></span>
<span><span class="co">#&gt; # A tibble: 9 × 14</span></span>
<span><span class="co">#&gt;   reg_id reg_chr reg_start reg_end mut_id patient_id chr   start   end gene </span></span>
<span><span class="co">#&gt;   &lt;chr&gt;  &lt;chr&gt;       &lt;int&gt;   &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;      &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;</span></span>
<span><span class="co">#&gt; 1 r1     1            1000    1000 m01    p1         1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 2 r1     1            1000    1000 m01    p2         1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 3 r1     1            1000    1000 m01    p1         1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 4 r1     1            1000    1000 m01    p2         1      1000  1000 g1   </span></span>
<span><span class="co">#&gt; 5 r3     2            3000    3000 m03    p1         2      3000  3000 g2   </span></span>
<span><span class="co">#&gt; 6 r1     1            1000    1000 sv01   p4         &lt;NA&gt;     NA    NA &lt;NA&gt; </span></span>
<span><span class="co">#&gt; 7 r3     2            3000    3000 &lt;NA&gt;   &lt;NA&gt;       &lt;NA&gt;     NA    NA &lt;NA&gt; </span></span>
<span><span class="co">#&gt; 8 &lt;NA&gt;   &lt;NA&gt;           NA      NA &lt;NA&gt;   p3         &lt;NA&gt;     NA    NA &lt;NA&gt; </span></span>
<span><span class="co">#&gt; 9 &lt;NA&gt;   &lt;NA&gt;           NA      NA &lt;NA&gt;   p5         &lt;NA&gt;     NA    NA &lt;NA&gt; </span></span>
<span><span class="co">#&gt; # ℹ 4 more variables: bp1_chr &lt;chr&gt;, bp1_pos &lt;dbl&gt;, bp2_chr &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   bp2_pos &lt;dbl&gt;</span></span></code></pre></div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing paneldesign</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jonas Ibn-Salem <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0001-9886-113X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/TRON-Private/paneldesign/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/TRON-Private/paneldesign/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Jonas Ibn-Salem.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</div>






  </body>
</html>
