<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en-US"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Get all exons as GRanges object — get_exons • paneldesign</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Get all exons as GRanges object — get_exons"><meta property="og:description" content="Get all exons as GRanges object"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">paneldesign</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Get all exons as GRanges object</h1>

    <div class="hidden name"><code>get_exons.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Get all exons as GRanges object</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_exons</span><span class="op">(</span><span class="va">edb</span>, columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"tx_id"</span>, <span class="st">"gene_id"</span>, <span class="st">"symbol"</span>, <span class="st">"gene_name"</span><span class="op">)</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-edb">edb<a class="anchor" aria-label="anchor" href="#arg-edb"></a></dt>
<dd><p>An Ensbel database object (EnsDb for Ensembl), e.g. from EnsDb.Hsapiens.v75</p></dd>


<dt id="arg-columns">columns<a class="anchor" aria-label="anchor" href="#arg-columns"></a></dt>
<dd><p>Annotation columns. See <code><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">ensembldb::exons()</a></code></p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional paramters passed to <code><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">ensembldb::exons()</a></code></p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A GenomicRanges::GRanges() object with regions.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="va">EnsDb.Hsapiens.v75</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: EnsDb.Hsapiens.v75</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: ensembldb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: generics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘generics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.difftime, as.factor, as.ordered, intersect, is.element, setdiff,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     setequal, union</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, is.unsorted, lapply,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     mapply, match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rank, rbind, rownames, sapply, saveRDS, table, tapply, unique,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:utils’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     findMatches</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicFeatures</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationDbi</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>replacing previous import ‘S4Arrays::makeNindexFromArrayViewport’ by ‘DelayedArray::makeNindexFromArrayViewport’ when loading ‘SummarizedExperiment’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: AnnotationFilter</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: 'ensembldb'</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from 'package:stats':</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     filter</span>
<span class="r-in"><span><span class="va">edb</span> <span class="op">&lt;-</span> <span class="va">EnsDb.Hsapiens.v75</span></span></span>
<span class="r-in"><span><span class="fu">get_exons</span><span class="op">(</span><span class="va">edb</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>cannot switch some GRCh37's seqlevels from NCBI to UCSC style</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 1314240 ranges and 5 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   seqnames            ranges strand |           tx_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00002234944     chr1       11869-12227      + | ENST00000456328</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00002234632     chr1       11872-12227      + | ENST00000515242</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00002269724     chr1       11874-12227      + | ENST00000518655</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001948541     chr1       12010-12057      + | ENST00000450305</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001671638     chr1       12179-12227      + | ENST00000450305</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ...      ...               ...    ... .             ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001741452     chrY 28774418-28774584      - | ENST00000435945</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001681574     chrY 28776794-28776896      - | ENST00000435945</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001638296     chrY 28779492-28779578      - | ENST00000435945</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001797328     chrY 28780670-28780799      - | ENST00000435945</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001794473     chrY 59001391-59001635      + | ENST00000431853</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           gene_id      symbol   gene_name         exon_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       &lt;character&gt; &lt;character&gt; &lt;character&gt;     &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00002234944 ENSG00000223972     DDX11L1     DDX11L1 ENSE00002234944</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00002234632 ENSG00000223972     DDX11L1     DDX11L1 ENSE00002234632</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00002269724 ENSG00000223972     DDX11L1     DDX11L1 ENSE00002269724</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001948541 ENSG00000223972     DDX11L1     DDX11L1 ENSE00001948541</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001671638 ENSG00000223972     DDX11L1     DDX11L1 ENSE00001671638</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ...             ...         ...         ...             ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001741452 ENSG00000237917     PARP4P1     PARP4P1 ENSE00001741452</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001681574 ENSG00000237917     PARP4P1     PARP4P1 ENSE00001681574</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001638296 ENSG00000237917     PARP4P1     PARP4P1 ENSE00001638296</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001797328 ENSG00000237917     PARP4P1     PARP4P1 ENSE00001797328</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001794473 ENSG00000235857     CTBP2P1     CTBP2P1 ENSE00001794473</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 273 sequences (1 circular) from 2 genomes (hg19, GRCh37)</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get exons for specifc genes:</span></span></span>
<span class="r-in"><span><span class="fu">get_exons</span><span class="op">(</span><span class="va">edb</span>, filter <span class="op">=</span> <span class="fu">SymbolFilter</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TP53"</span>, <span class="st">"BRAF"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 178 ranges and 5 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   seqnames              ranges strand |           tx_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                      &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001657961    chr17     7565097-7565332      - | ENST00000413465</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00002204316    chr17     7569404-7569562      - | ENST00000359597</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003492844    chr17     7571720-7573008      - | ENST00000504290</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003492844    chr17     7571720-7573008      - | ENST00000510385</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003492844    chr17     7571720-7573008      - | ENST00000504937</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ...      ...                 ...    ... .             ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003603715     chr7 140549911-140550012      - | ENST00000497784</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003470674     chr7 140549911-140550012      - | ENST00000469930</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001154485     chr7 140624366-140624564      - | ENST00000288602</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001862791     chr7 140624366-140624458      - | ENST00000497784</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001888191     chr7 140624366-140624509      - | ENST00000469930</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                           gene_id      symbol   gene_name         exon_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                       &lt;character&gt; &lt;character&gt; &lt;character&gt;     &lt;character&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001657961 ENSG00000141510        TP53        TP53 ENSE00001657961</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00002204316 ENSG00000141510        TP53        TP53 ENSE00002204316</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003492844 ENSG00000141510        TP53        TP53 ENSE00003492844</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003492844 ENSG00000141510        TP53        TP53 ENSE00003492844</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003492844 ENSG00000141510        TP53        TP53 ENSE00003492844</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ...             ...         ...         ...             ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003603715 ENSG00000157764        BRAF        BRAF ENSE00003603715</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00003470674 ENSG00000157764        BRAF        BRAF ENSE00003470674</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001154485 ENSG00000157764        BRAF        BRAF ENSE00001154485</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001862791 ENSG00000157764        BRAF        BRAF ENSE00001862791</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   ENSE00001888191 ENSG00000157764        BRAF        BRAF ENSE00001888191</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 2 sequences from hg19 genome</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jonas Ibn-Salem.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

